app/entity-repositories/pasteRepository.js

```diff
@@ -0,0 +1,101 @@
+class PasteRepository {
+    constructor ({ dbProvider }) {
+        this.dbProvider = dbProvider;
+    }
+
+    async findByID({ id }) {
+
+        const result = await this.findOne({ id });
+        return result;
+
+    }
+
+    async findOne({ id, name, authorID }) {
+        
+        const result = await this.findAll({ id, name, authorID });
+        
+        if(!result) {
+            return null;
+        }
+
+        return result[0];
+
+    }
+
+    async findAll({ id, name, authorID}) {
+
+        const itemsToFind = [];
+
+        if ( id ) {
+
+            const arrayOfIDs = [];
+
+            if (Array.isArray(id)) {
+                
+                for (let i = 0; i < id.length; i++) {
+                    arrayOfIDs.push(id[i]);
+                }
+
+                itemsToFind.push(`id in (${arrayOfIDs.join(', ')})`);
+
+            } else {
+
+                itemsToFind.push(`id in (${id})`);
+
+            }
+        }
+
+        if( name ) {
+            itemsToFind.push(`name='${name}'`);
+        }
+
+        if( authorID ) {
+            itemsToFind.push(`author_id=${authorID}`);
+        }
+        
+        //console.log(`SELECT * FROM paste WHERE ${itemsToFind.join(' AND ')}`);
+        const result = await this.dbProvider.execute(`SELECT * FROM paste WHERE ${itemsToFind.join(' AND ')}`);
+
+        if(!result) {
+            return null;
+        }
+
+        return result;
```

Comment:
And here we need to construct actual entity out of items we found
(yeah, using `new Paste(....)`).
Consider of making some kind of `PasteFactory` class (e.g. search for `factory pattern` via google for details)

Reply:
All of comments for this repo is appliable for any another one.